internal/core/types.go

```diff
@@ -78,12 +78,40 @@ type UpdateRoomOptions struct {
 }
 
 // SendMessageOptions contains parameters to pass when sending a new message.
-type SendMessageOptions struct {
+type SendMessageOptions = SendSimpleMessageOptions
+
+// SendMultipartMessageOptions contains parameters to pass when sending a new message.
+type SendMultipartMessageOptions struct {
+	RoomID   string
+	SenderID string
+	Parts    []NewPart
+}
+
+// SendSimpleMessageOptions contains parameters to pass when sending a new message.
+type SendSimpleMessageOptions struct {
 	RoomID   string
 	Text     string
 	SenderID string
 }
 
+type NewPart interface {
```

Comment:
Can we not name this interface just `Part` and then the method `isPart()`?

Reply:
This is for [conformance with the spec](https://github.com/pusher/chatkit-spec/blob/master/client-sdks/messages.md#create-a-message-1) and the other SDKs in theory... but actually now that you mention it neither Android or Swift are following the same convention 😞 (partly my bad since that's definitely the kind of thing I should have been looking for in a review... and I think I looked at both yours and @mdpye's PRs).